RocketChat/Rocket.Chat
# fix: New users are not automatically inserted in auto-join rooms inside default teams

## Proposed changes (including videos or screenshots)


Earlier, whenever we had auto join teams, when the user joined, we would not look at the auto-join channels inside of it.
The idea is, that when we are going to create the subscriptions between the team and the user, we check whether the team has a default channel, if so, fetch those and add the user to it as well

## Issue(s)


## Steps to test or reproduce

1. Convert the general room into a team
2. Create a new channel inside the general team
3. Set the new channel as auto-join
4. Create a new user

Expected Behavior: The user should be added to both the team and the channel
Current behavior: The user is being added only to the team

## Further comments

CORE-58
